src/System.Drawing.Common/src/System/Drawing/Icon.Unix.cs

```diff
@@ -214,7 +214,7 @@ public Icon(Icon original, Size size)
                 }
 
                 if (id == ushort.MaxValue)
-                    throw new ArgumentException("Icon", "No valid icon image found");
+                    throw new ArgumentException(SR.NoValidIconImageFound, "Icon");
```

Comment:
Should this instead be?
```suggestion
                    throw new ArgumentException(SR.NoValidIconImageFound, nameof(original));
```

Reply:
Also in the `Icon(Type, string)` ctor there is a `throw ArgumentException(null)` when the resource cannot be found that should probably be: `throw new ArgumentException(SR.Format(SR.ResourceNotFound, type, resource));` as is used in the Windows code:
https://github.com/dotnet/corefx/blob/41cd99d051102be4ed83f4f9105ae9e73aa48b7c/src/System.Drawing.Common/src/System/Drawing/Icon.Windows.cs#L105-L111